Cargo.toml

```diff
@@ -11,16 +11,14 @@ edition = "2021"
 
 [dependencies]
 bitcoin = { version = "0.31", features = ["rand"] }
+getrandom = { version = "0.2", features = ["custom"] }
```

Comment:
This is not enough to make it work on IC. Use the `canister-sdk/ic-crypto-getrandom-for-wasm` instead.

Reply:
Using `canister-sdk/ic-crypto-getrandom-for-wasm` did not solve the problem. Perhaps another platform-related issue? Looking into it now.

Reply:
The correct way to fix the issue is

```toml
getrandom = { workspace = true, features = ["custom"] }
```

and then in lib.rs

/// GetRandom fixup to allow getrandom compilation.
/// A getrandom implementation that always fails
///
/// This is a workaround for the fact that the `getrandom` crate does not compile
/// for the `wasm32-unknown-ic` target. This is a dummy implementation that always
/// fails with `Error::UNSUPPORTED`.
pub fn getrandom_always_fail(_buf: &mut [u8]) -> Result<(), getrandom::Error> {
    Err(getrandom::Error::UNSUPPORTED)
}

getrandom::register_custom_getrandom!(getrandom_always_fail);

Reply:
or canister-sdk/ic-crypto-getrandom-for-wasm  I guess...

Reply:
But in general the dependency should be set only for wasm32 arch, this library shouldn't be available only for canisters, but for any kind of arch

Reply:
> The correct way to fix the issue is
> 
> ```toml
> getrandom = { workspace = true, features = ["custom"] }
> ```
> 
> and then in lib.rs
> 
> ```rust
> /// GetRandom fixup to allow getrandom compilation.
> /// A getrandom implementation that always fails
> ///
> /// This is a workaround for the fact that the `getrandom` crate does not compile
> /// for the `wasm32-unknown-ic` target. This is a dummy implementation that always
> /// fails with `Error::UNSUPPORTED`.
> pub fn getrandom_always_fail(_buf: &mut [u8]) -> Result<(), getrandom::Error> {
>     Err(getrandom::Error::UNSUPPORTED)
> }
> 
> getrandom::register_custom_getrandom!(getrandom_always_fail);
> ```

This was the first workaround I tried but it doesn't work either.

Reply:
that's weird, because I'm sure it builds canisters with that. Maybe another issue with M1?